docs/DeveloperGuide.md

```diff
@@ -1,38 +1,360 @@
+module name.
+
+The following sequence diagram shows how the `findModuleByName` operation works:
+
+![FindModule](diagrams/FindModule.png)
+
+<small><i>Figure ???</i></small>
+
+### 3.5. Delete Modules
+
+`delete` command:
+
+The `delete` command is used by the user to delete a module which is specified by the user.
+
+The sequence in which the `Parser` class handles the `delete` command is as follows:
+1) The `Parser` class extracts the necessary fields from the user input, and calls upon the `deleteModule`
+   method in the `ModuleList` class.
+2) The method then loops through the moduleList array, attempting to find a module with the module code
+   that was specified by the user.
+3) If a corresponding module is found, the `deleteModule` method will remove the module from the moduleList array
+   and return the deleted `Module` object. Otherwise, the `deleteModule` method returns `null`.
+4) If a deleted `Module` object is returned from `deleteModule`, the `Parser` class calls upon
+   `printDeletedModule()` of the `Print` class. Otherwise, the `Parser` class calls upon `printNoDeletedModuleFound()`
+   of the `Print` class. This displays to the user the result of the `delete` command.
+
+![deleteModule](diagrams/DeleteModule.png)
+
+<small><i>Figure ???</i></small>
+
+### 3.6. Edit Modules
+
+`edit` command:
+
+The `edit` command is used by the user to make changes to fields in the module description.
+
+The sequence in which the `Parser` class handles the `edit` command is as follows:
+1) The `Parser` class extracts the other fields of the user input, and calls upon the `editModuleField()` method
+   by self-invocation.
+2) The method then uses a `switch` statement to evaluate the module field that the user wants to update.
+3) After checking the field to be updated, the appropriate method within the `ModuleList` class is called upon to
+   make the changes specified by the user.
+4) If the user wants to update Modular Credits, the `editModularCredits()` method is called.
+   This method directly modifies the `modularCredits` attribute of the `Module` object.
+5) If the user wants to update Module Type, the `editModuleType()` method is called.
+   This method removes the existing `Module` object and adds a new `Module` object of the new type specified
+   by the user. The `getModuleCode()`, `getModularCredits()`, `getYear()`, `getSemester()` and `getGrade()`
+   methods in the `Module` class are also called upon to obtain the respective fields required to create the new
+   `Module` object.
+6) If the user wants to update the Year, the `editYear()` method is called.
+   This method directly modifies the `year` attribute of the `Module` object.
+7) If the user wants to update the Semester, the `editSemester()` method is called.
+   This method directly modifies the `semester` attribute of the `Module` object.
+
+![editCommand](diagrams/EditCommand.png)
```

Comment:
<img width="77" alt="image" src="https://user-images.githubusercontent.com/67412977/228729293-be8b35b8-9e64-4f78-b2e8-dbbdbd7148f7.png">
Add bars for self invocaton